client-java/src/main/java/uk/gov/gchq/palisade/client/internal/request/PalisadeResponse.java

```diff
@@ -26,35 +26,48 @@
 
 /**
  * An instance of {@link PalisadeResponse} is returned after a successful
- * request to the Palisade Service. This object contains the url and the token
- * which is to be used to contact the Filtered Resource Service and wait for
- * resources to become available.
+ * request to the Palisade Service.
  * <p>
- * Note that the {@link PalisadeResponse} class is created at compile time. The
- * way in which the class is created is determined by the
- * {@code ImmutableStyle}. This class is also compatible with Jackson.
+ * Note that the {@link ImmutablePalisadeResponse} implementation class is
+ * created at compile time. The way in which the class is created is determined
+ * by the {@code ImmutableStyle}. The generated class is package private hence
+ * the nested {@code Builder}.
+ * <p>
+ * This class is also compatible with Jackson.
  *
  * @see "https://immutables.github.io/style.html"
  * @since 0.5.0
  */
 @Value.Immutable
 @ImmutableStyle
-@JsonDeserialize(as = PalisadeResponse.class)
-@JsonSerialize(as = PalisadeResponse.class)
-public interface IPalisadeResponse extends Serializable {
+@JsonDeserialize(as = ImmutablePalisadeResponse.class)
+@JsonSerialize(as = ImmutablePalisadeResponse.class)
```

Comment:
That's there so that testing is far easier. It allows creating the object and then serialising it to JSON without having to manually create and embed the JSON in the test which is error prone with all the escaping required. This would be so much easy with multi-line strings.